---
title: "Detect Batch effect Report 1"
author: "Constanza Silva"
date: "July 12, 2017"
output: html_document
---
#Wednesday Bootcamp Report 1

```{r setup, include=FALSE}
# To run DataCamp tutorials on your computer, install the packages below

#install.packages("nasaweather")
#install.packages("dplyr")
#install.packages("ggvis")

# To run the batch effects case study, install the packages below
source("http://bioconductor.org/biocLite.R")
#biocLite("Biobase")
#biocLite("genefilter")
devtools::install_github("genomicsclass/GSE5859")
```

  Perform the analysis in the R code below (part 1 only) and show your analysis in an R markdown file, named batch-effect-part1.rmd. Enhance the presentation of your R markdown if possible.

  (Optional) Perform the analysis in the R code below (part 2 only) and show your analysis in another R markdown file, named batch-effect-part2.rmd.

  Publish your analysis using workflowr.

```{r}
library(Biobase)
#install.packages("rafalib")
library(rafalib)
library(genefilter)
library(GSE5859)
data(GSE5859)
```

## gene expression levels  
```{r}
geneExpression = exprs(e)
```

## print date info
```{r}
sampleInfo = pData(e)
head(sampleInfo$date)
```

## Output: print a table of year vs ethnic groups
```{r}
year = factor( format(sampleInfo$date,"%y") ) #format () function is to statify date using in this case "data" as a variable.
#stratify the data by data they were colected. The majority of the Asian samples (ASN) were collected on 2005.
tab = table(year,sampleInfo$ethnicity)
print(tab)
```
The majority were collected on 2005.The data has a batch effect on differential expression between Asians and Europeans.

## Differential expression analysis
```{r}
##remove control genes
out <- grep("AFFX",rownames(geneExpression))

# Use a t-test to perform differential expression analysis 
# between CEU and ASN
eth <- sampleInfo$ethnicity
ind <- which(eth%in%c("CEU","ASN")) #CEU are Europeans
res1 <- rowttests(geneExpression[-out,ind],droplevels(eth[ind]))

# Use a t-test to perform differential expression analysis 
# between 2002 vs 2003 within CEU
ind <- which(year%in%c("02","03") & eth=="CEU")
res2 <- rowttests(geneExpression[-out,ind],droplevels(year[ind])) #t.test by row rowttest()

# output: plot histograms for p-values 
hist(res1$p, main="p-value of CEU vs ASN")
hist(res2$p, main="p-value of 2002 vs 2003 for CEU") #The distribution of the p-values (to the left) shows there is a bath effect within CE sample regarding the date of sampling. 
```

The distribution of the p-values (to the left) shows there is a bath effect within CE sample regarding the date of sampling. 